src/tests/ftest/util/test_utils.py

```diff
@@ -223,21 +269,91 @@ def create(self):
+            # Call the dmg pool create command
+            self.dmg.action.value = "create"
+            self.dmg.get_action_command()
+            # uid/gid used in API correspond to --user and --group in dmg.
+            # group, or self.name.value, used in API is called server group and
+            # it's different from the group name passed in to --group. Server
+            # group isn't used in dmg. We don't pass it into the command, but
+            # we'll still use it to set self.pool.group
+            self.dmg.action_command.group.value = self.groupname.value
+            self.dmg.action_command.user.value = self.username.value
+            self.dmg.action_command.scm_size.value = self.scm_size.value
+            self.dmg.action_command.ranks.value = ranks_comma_separated
+            self.dmg.action_command.nsvc.value = self.svcn.value
+            create_result = self.dmg.run()
+            self.log.info("Result stdout = %s", create_result.stdout)
+            self.log.info("Result exit status = %s", create_result.exit_status)
+            # Get UUID and service replica from the output
+            uuid_svc = get_pool_uuid_service_replicas_from_stdout(
+                create_result.stdout)
+            new_uuid = uuid_svc[0]
+            service_replica = uuid_svc[1]
+
+            # 3. Create DaosPool object. The process is similar to the one in
+            # DaosPool.create, but there are some modifications
+            if self.name.value is None:
+                self.pool.group = None
+            else:
+                self.pool.group = ctypes.create_string_buffer(self.name.value)
+            # Modification 1: Use the length of service_replica returned by dmg
+            # to calculate rank_t. Note that we assume we always get a single
+            # number. I'm not sure if we ever get multiple numbers, but in that
+            # case, we need to modify this implementation to create a list out
+            # of the multiple numbers possibly separated by comma
+            service_replicas = [int(service_replica)]
+            rank_t = ctypes.c_uint * len(service_replicas)
+            # Modification 2: Use the service_replicas list to generate rank.
+            # In DaosPool, we first use some garbage 999999 values and let DAOS
+            # set the correct values, but we can't do that here, so we need to
+            # set the correct rank value by ourself
+            rank = rank_t(*list([svc for svc in service_replicas]))
+            rl_ranks = ctypes.POINTER(ctypes.c_uint)(rank)
+            # Modification 3: Similar to 1. Use the length of service_replicas
+            # list instead of self.svcn.value
+            self.pool.svc = daos_cref.RankList(rl_ranks, len(service_replicas))
+
+            # 4. Set UUID and attached to the DaosPool object
+            self.pool.set_uuid_str(new_uuid)
+            self.pool.attached = 1
+        else:
+            self.log.error("Cannot create pool! Use USE_API or USE_DMG")
```

Comment:
Do we expect to have more than two methods for pool creation?  If not, should we use a boolean instead to avoid the need for this error?

Reply:
I don't think we'll have more than two in the near future, so I removed this condition.